Remotion/Web/ClientScript/Scripts/DropDownMenu.ts

```diff
@@ -0,0 +1,928 @@
+        {
+          if (currentItemIndex < itemInfos!.length - 1) // TODO: nullcheck above
+          {
+            currentItemIndex++;
+            if (currentItemIndex === 1 && oldIndex === -1)
+              oldIndex = 0;
+          }
+          else
+          {
+            currentItemIndex = 0;
+          }
+
+          if (oldIndex === currentItemIndex)
+            break;
+          if ($(dropDownMenuItems[currentItemIndex]).children('a').length > 0) // skip separators
+            break;
+        }
+        break;
+      case 38: // up arrow
+        event.preventDefault();
+        event.stopPropagation();
+        isSelectionUpdated = true;
+        oldIndex = currentItemIndex >= 0 ? currentItemIndex : itemInfos!.length - 1; // TODO: nullcheck above
+        while (true) // skip separators
+        {
+          if (currentItemIndex > 0)
+            currentItemIndex--;
+          else
+            currentItemIndex = itemInfos!.length - 1; // TODO: nullcheck above
+
+          if (oldIndex === currentItemIndex)
+            break;
+          if ($(dropDownMenuItems[currentItemIndex]).children('a').length > 0)
+            break;
+        }
+        break;
+    }
+
+    if (isSelectionUpdated)
+    {
+      $("li", DropDownMenu._currentPopup).removeClass (DropDownMenu._itemSelectedClassName);
+      if (currentItemIndex >= 0 && currentItemIndex < itemInfos!.length) // TODO: nullcheck above
+      {
+        var dropDownMenuItem = $(dropDownMenuItems[currentItemIndex]);
+        if (dropDownMenuItem.children('a').length > 0) // skip separators
+        {
+          dropDownMenuItem.addClass(DropDownMenu._itemSelectedClassName);
+          var nonJQueryAnchor = dropDownMenuItem.children ('a')[0];
+          // do not use jQuery focus() here because jQuery causes the focus/blur events to fire in the wrong order (first focus, then blur)
+          nonJQueryAnchor.focus();
+        }
+      }
+    }
+  }
+
+  private static GetItemEnabled (itemInfo: DropDownMenu_ItemInfo, selectionCount: number): boolean
+  {
+    var isEnabled = true;
+    if (itemInfo.IsDisabled)
+    {
+      isEnabled = false;
+    }
+    else
+    {
+      if (itemInfo.RequiredSelection == DropDownMenu._requiredSelectionExactlyOne
+          && selectionCount != 1)
+      {
+        isEnabled = false;
+      }
+      if (itemInfo.RequiredSelection == DropDownMenu._requiredSelectionOneOrMore
+          && selectionCount < 1)
+      {
+        isEnabled = false;
+      }
+    }
+    return isEnabled;
+  }
+
+  // TODO: figure out exact types
+  private static GetTarget (event: JQueryMouseEventObject, tagName: string): Node
+  {
+    var element = event.target as Node & Element;
+    while (element && element.tagName != tagName)
+      element = element.parentNode as Node & Element;
+    // more fun with IE, sometimes event.target is empty, just ignore it then
+    if (!element)
+      return [] as unknown as Node;
+    return element;
+  }
```

Comment:
```suggestion
  private static GetTarget (event: JQueryMouseEventObject, tagName: string): Element
  {
    var element: Nullable<Element> = event.target;
    while (element && element.tagName != tagName)
      element = element.parentNode as Nullable<Element>;
    // more fun with IE, sometimes event.target is empty, just ignore it then
    if (!element) // TODO: What is this? This should also fail by the caller.
      return [] as unknown as Element;
    return element; // TODO: Handle parent possible being null
  }
```

I don't know why this is like it is but the signature should be good, the code inside is just broken.